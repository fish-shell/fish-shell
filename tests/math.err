
####################
# Validate basic expressions

####################
# Validate some rounding functions

####################
# Validate some integral computations

####################
# Validate how variables in an expression are handled

####################
# Validate math error reporting
math: Error: Too few arguments
'2 - '
    ^
math: Error: Too few arguments
'ncr(1)'
      ^
math: Error: Expression is bogus
'max()'
   ^
math: Error: Too few arguments
'sin()'
     ^
math: Error: Too many arguments
'2 + 2 4'
       ^
math: Expected at least 1 args, got only 0
math: Expected at least 1 args, got only 0
math: Error: Result is infinite
'2^999999'
math: Error: Result is infinite
'1 / 0'

####################
# Validate x as multiplier
math: Error: Unknown variable
'2 x4'
    ^
