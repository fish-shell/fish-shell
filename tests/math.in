math 3 / 2
math 10/6
math -s0 10 / 6
math -s3 10/6
math '10 % 6'
math -s0 '10 % 6'
math '23 % 7'
math -s6 '5 / 3 * 0.3'
true
