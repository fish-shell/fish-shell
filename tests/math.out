1
1
1
1.666
4
4
2
.499999
